docs/common/REACTIVE_TERNARY_AND_STYLE_PITFALLS.md

```diff
@@ -0,0 +1,223 @@
+# Reactive Ternary and Style Pitfalls
+
+**TL;DR:** Reactive values (derived/cell) are proxies that are always truthy in ternaries. Layout style properties (`justifyContent`, `flexDirection`) don't work with reactive values. Use `ifElse` with static styles instead.
+
+## Quick Reference
+
+```tsx
+// BAD - ternary with reactive value (proxy always truthy)
+const result = reactiveBoolean ? "yes" : "no";  // ALWAYS "yes"
+
+// BAD - reactive value in layout style
+<div style={{ justifyContent: reactiveDerived }}>  // Won't work
+
+// GOOD - use ifElse with static styles
+ifElse(reactiveBoolean,
+  <div style={{ justifyContent: "flex-end" }}>content</div>,
+  <div style={{ justifyContent: "flex-start" }}>content</div>
+)
+```
+
+---
+
+## Background
+
+This document captures pitfalls discovered while developing a chat pattern where messages needed conditional alignment:
+- **My messages**: Blue bubbles, RIGHT-aligned
+- **Other's messages**: Grey bubbles, LEFT-aligned with avatar
+- **System messages**: Centered, italic
+
+## What Didn't Work
+
+### Attempt 1: Ternary Operators with Reactive Values
+
+```tsx
+// BROKEN - proxy is always truthy
+const isMyMessage = derive(meta, m => m?.isMyMessage ?? false);
+
+<div style={{
+  flexDirection: isMyMessage ? "row-reverse" : "row"  // Always evaluates to "row-reverse"
+}}>
+```
+
+**Why it fails:** Derived values are reactive proxies. In JavaScript, any object (including proxies) is truthy. So `isMyMessage ? x : y` ALWAYS returns `x`, regardless of the underlying boolean value.
```

Comment:
This is surprising to me... I was under the impression we would automatically transform the ternary here to an `ifElse` which _should_ work, e.g.

```tsx
const isMyMessage = derive(meta, m => m?.isMyMessage ?? false);

<div style={{
  flexDirection: ifElse(isMyMessage, "row-reverse", "row")
}}>
```

Perhaps we have a compiler bug?